content/blog/fast-data-processing-pipeline-for-predicting-flight-delays-using-apache-apis-kafka-spark-streaming-and-machine-learning-part-2.md

```diff
@@ -0,0 +1,231 @@
+
+***Editor's Note: You can find Part 1 of this series [here](https://developer.hpe.com/blog/fast-data-processing-pipeline-for-predicting-flight-delays-using-apache-/)***
+
+According to Bob Renner, CEO of [Liaison Technologies](https://www.liaison.com/), the possibility to blend machine learning with real-time transactional data flowing through a single platform is opening a world of new possibilities, such as enabling organizations to take advantage of opportunities as they arise. Leveraging these opportunities requires fast, scalable data processing pipelines that process, analyze, and store events as they arrive.
+
+This is the second in a series of blogs, which discusses the architecture of a data pipeline that combines streaming data with machine learning and fast storage. [The first post](https://www.hpe.com/us/en/software/data-fabric.html) discussed creating a machine learning model to predict flight delays. This second post will discuss using the saved model with streaming data to do real-time analysis of flight delays. The third post will discuss fast storage with MapR Database.
+
+![Saved model with streaming data](/img/saved-model-with-streaming-data.png)
+
+## Microservices, Data Pipelines, and Machine Learning Logistics
+
+The [microservice architectural style](https://martinfowler.com/articles/microservices.html) is an approach to developing an application as a suite of small independently deployable services. A common architecture pattern combined with microservices is event sourcing using an append-only publish-subscribe event stream such as MapR Event Streams (which provides a Kafka API).
+
+![Immediate access to operational and analytical data in MapR](/img/immediate-access-to-operational-analytical-data.png)
+
+## Publish Subscribe Event Streams with MapR Event Store
+
+A central principle of publish/subscribe systems is decoupled communications, wherein producers don’t know who subscribes, and consumers don’t know who publishes; this system makes it easy to add new listeners or new publishers without disrupting existing processes. MapR Event Store allows any number of information producers (potentially millions of them) to publish information to a specified topic. MapR Event Store will reliably persist those messages and make them accessible to any number of subscribers (again, potentially millions). Topics are partitioned for throughput and scalability, producers are load balanced and consumer can be grouped to read in parallel. MapR Event Store can scale to very high throughput levels, easily delivering millions of messages per second using very modest hardware.
+
+![Kafka API](/img/kafka-api.png)
+
+You can think of a partition like a queue; new messages are appended to the end, and messages are delivered in the order they are received.
+
+![MapR Cluster](/img/mapr-cluster.png)
```

Comment:
The images also seem very large. This causes an issue for some of them where they appear fuzzy. You might want to adjust the size of the images. This can be done either by adjust the image itself or in the markdown through code. (see the guide for more info)

Reply:
I've updated all the images to have a width of 700.

Reply:
This part of the documentation is confusing:
<img width="970" alt="Screen Shot 2021-11-03 at 10 28 36 PM" src="https://user-images.githubusercontent.com/19980501/140262836-494ae6ba-f074-4dac-afd2-5e1174b2bcdf.png">
I feel like the instructions are telling me to delete the img element that I just added.

Reply:
Also, the instructions do not account for img alt tags, which I would recommend.